src/Sylius/Bundle/PayumBundle/DependencyInjection/Compiler/RegisterGatewayConfigTypePass.php

```diff
@@ -47,8 +51,15 @@ public function process(ContainerBuilder $container): void
             }
         }
 
-        $gatewayFactories = array_merge($gatewayFactories, ['offline' => 'sylius.payum_gateway_factory.offline']);
-        ksort($gatewayFactories);
+
+        usort($gatewayFactories, function (array $firstGateway, array $secondGateway): int {
+            return $secondGateway['priority'] - $firstGateway['priority'];
+        });
+
+        foreach ($gatewayFactories as $key => $factory) {
+            $gatewayFactories[$factory['type']] = $factory['label'];
+            unset($gatewayFactories[$key]);
+        }
 
         $container->setParameter('sylius.gateway_factories', $gatewayFactories);
```

Comment:
Could we add an integration test for this compiler pass as we have similar ones for other passes? This way we can ensure it's sorted correctly.